build-system/tasks/visual-diff.js

```diff
@@ -35,10 +100,463 @@ function setPercyBranch() {
+ *     eventually appear on the page.
+ */
+async function verifyCssElements(page, url, forbiddenCss, loadingIncompleteCss,
+  loadingCompleteCss) {
+  // Wait for loader dot to be hidden.
+  await page.waitForSelector('.i-amphtml-loader-dot', {
+    hidden: true,
+    timeout: CSS_SELECTOR_TIMEOUT_MS,
+  }).catch(() => {
+    log('fatal', colors.cyan(url),
+        `still has the AMP loader dot after ${CSS_SELECTOR_TIMEOUT_MS} ms`);
+  });
+
+  if (forbiddenCss) {
+    for (const css of forbiddenCss) {
+      if (await page.$(css) !== null) {
+        log('fatal', colors.cyan(url), 'has forbidden CSS element',
+            colors.cyan(css));
+      }
+    }
+  }
+
+  if (loadingIncompleteCss) {
+    for (const css of loadingIncompleteCss) {
+      log('verbose', `Waiting for invisibility of ${css}`);
+      await page.waitForSelector(css, {
+        hidden: true,
+        timeout: CSS_SELECTOR_TIMEOUT_MS,
+      }).catch(() => {
+        log('fatal', colors.cyan(url), 'still has CSS element',
+            colors.cyan(css), `after ${CSS_SELECTOR_TIMEOUT_MS} ms`);
+      });
+    }
+  }
+
+  if (loadingCompleteCss) {
+    for (const css of loadingCompleteCss) {
+      log('verbose', 'Waiting for visibility of', css);
+      await page.waitForSelector(css, {
+        visible: true,
+        timeout: CSS_SELECTOR_TIMEOUT_MS,
+      }).catch(() => {
+        log('fatal', colors.cyan(url), 'does not yet have CSS element',
+            colors.cyan(css), `after ${CSS_SELECTOR_TIMEOUT_MS} ms`);
+      });
+    }
+  }
+}
+
+/**
+ * Enables the given AMP experiments.
+ *
+ * @param {!puppeteer.Page} page a Puppeteer control browser tab/page.
+ * @param {*} experiments List of experiments to enable.
+ */
+async function enableExperiments(page, experiments) {
+  if (experiments) {
+    log('verbose', 'Setting AMP experiments',
+        colors.cyan(experiments.join(', ')));
+    await page.setCookie({
+      name: 'AMP_EXP',
+      value: experiments.join('%2C'),
+      domain: HOST,
+    });
+  }
+}
+
+/**
+ * Clears all AMP experiment cookies.
+ *
+ * @param {!puppeteer.Page} page a Puppeteer control browser tab/page.
+ */
+async function clearExperiments(page) {
+  await page.deleteCookie({name: 'AMP_EXP', domain: HOST});
+}
+
+/**
+ * Enables debugging if requested via command line.
+ */
+function setDebuggingLevel() {
+  process.env.WEBSERVER_QUIET = '--quiet';
+
+  if (argv.debug) {
+    // eslint-disable-next-line google-camelcase/google-camelcase
+    argv.chrome_debug = true;
+    process.env.WEBSERVER_QUIET = '';
+  }
+  if (argv.webserver_debug) {
+    process.env.WEBSERVER_QUIET = '';
+  }
+}
+
+/**
+ * Creates a Percy build with only a blank page for comparison.
+ *
+ * Enables us to require percy checks on GitHub, and yet, not have to do a full
+ * build for every PR.
+ *
+ * @param {!puppeteer.Page} page a Puppeteer control browser tab/page.
+ */
+async function createEmptyBuild(page) {
```

Comment:
Once this is checked in and working, we should remember to flip the flag to make the Percy check mandatory once again :)